bin/check-instance-cpucredits.rb

```diff
@@ -0,0 +1,145 @@
+#   check-instance-cpucredits.rb
+#
+#   Check the specified instance's CPU credits and alert critical if below 10 and warn below 50 (must specify region):
+#   check-instance-cpucredits.rb -i i-0d2fe2a5f09338fef -r eu-west-1
+#
+#   Check instance CPU credits and warn if they are less than 100 and alert critical if below 10:
+#   check-instance-cpucredits.rb -w 100 -c 10
+#
+#   Check instance CPU credits:
+#   check-instance-cpucredits.rb -w 100 -c 10
+#
+# LICENSE:
+#    MIT License
+#
+#    Copyright (c) 2017 Claranet
+#
+#    Permission is hereby granted, free of charge, to any person obtaining a copy
+#    of this software and associated documentation files (the "Software"), to deal
+#    in the Software without restriction, including without limitation the rights
+#    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+#    copies of the Software, and to permit persons to whom the Software is
+#    furnished to do so, subject to the following conditions:
+#
+#    The above copyright notice and this permission notice shall be included in all
+#    copies or substantial portions of the Software.
+#
+#    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+#    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+#    SOFTWARE.
+#
+
+require 'net/http'
+require 'aws-sdk'
+require 'sensu-plugin/check/cli'
+
+class CheckInstanceCpuCredits < Sensu::Plugin::Check::CLI
+  VERSION = '0.0.1'.freeze
+
+  option :warning,
+         description: 'Issue a warning if the CPU credits for the instance fall below this value.',
+         short:       '-w N',
+         long:        '--warning VALUE',
+         default:     50
+
+  option :critical,
+         description: 'Issue a critical if the CPU credits for the instance fall below this value.',
+         short:       '-c N',
+         long:        '--critical VALUE',
+         default:     10
+
+  option :instance_id,
+         description: 'Instance ID to check the CPU credits for (optional)',
+         short:       '-i ID',
+         long:        '--instance-id ID'
+
+  option :region,
+         description: 'Instance ID to check the CPU credits for (optional)',
+         short:       '-r REGION',
+         long:        '--region REGION'
+
+  option :metadata_endpoint,
+         description: 'The AWS metadata server URL to use (optional)',
+         short:       '-m URL',
+         long:        '--metadata-url URL',
+         default:     'http://169.254.169.254/latest/meta-data/'
+
+  def my_instance_region
```

Comment:
can we move these to a shared lib? I am sure that more than this check could benefit from it.

Reply:
:+1: I will look into this.